core/src/main/resources/reference.conf

```diff
@@ -535,6 +535,41 @@ datastax-java-driver {
     #   change.
     # Overridable in a profile: no
     warn-on-init-error = true
+
+
+    advanced-shard-awareness {
+      # Whether to use advanced shard awareness when trying to open new connections.
+      #
+      # Having this enabled makes sense only for ScyllaDB clusters.
+      # Results in smaller connection storms in multi-client settings.
+      # If set to false the driver will not attempt to use this feature.
+      # If set to true the driver will attempt to use it and will log warnings each time something
+      # makes it not possible.
+      # If the node for some reason does not report it's sharding info the driver
+      # will log a warning and create connection the same way as if this feature was disabled.
+      # If the cluster ignores the request for specific shard warning will also be logged,
+      # although the local port will be chosen according to advanced shard awareness rules.
+      #
+      # Required: yes
+      # Modifiable at runtime: yes, the new value will be used for connections created after the
+      # change.
+      # Overridable in a profile: no
+      enabled = true
+
+      # Inclusive lower bound of port range to use in advanced shard awareness
+      # The driver will attempt to reserve ports for connection only within the range.
+      # Required: yes
+      # Modifiable at runtime: yes, the new value will be used for calls after the
+      # change.
+      port-low = 10000
+
+      # Inclusive upper bound of port range to use in advanced shard awareness.
+      # The driver will attempt to reserve ports for connection only within the range.
+      # Required: yes
+      # Modifiable at runtime: yes, the new value will be used for calls after the
+      # change.
+      port-high = 60000
```

Comment:
Why not 65535 ?

Reply:
Copied it from 3.x. It's likely just an arbitrary, high range. I can set it to 65535

Reply:
Set to 65535